File: app/src/main/java/com/decode/tumblr/dao/PhotoDao.java
package com.decode.tumblr.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.decode.tumblr.model.PhotoObject;

import java.util.List;

@Dao
public interface PhotoDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(PhotoObject photo);

    @Query("DELETE FROM photo_table")
    void deleteAll();

    @Query("SELECT * from photo_table ORDER BY id ASC")
    LiveData<List<PhotoObject>> getAllPhotos();

    @Query("SELECT * from photo_table WHERE id=:id")
    PhotoObject getById(int id);

}

File: app/src/main/java/com/decode/tumblr/data/db/dao/PhotoDao.kt
package com.decode.tumblr.data.db.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.decode.tumblr.data.db.model.PhotoObject

@Dao
interface PhotoDao {

    @get:Query("SELECT * from photo_table ORDER BY id ASC")
    val allPhotos: LiveData<List<PhotoObject>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(photo: PhotoObject): Long

    @Query("DELETE FROM photo_table")
    fun deleteAll()

    @Query("SELECT * from photo_table WHERE id=:id")
    fun getById(id: Int): PhotoObject

}
